src/Control/Lens/Setter.hs

```diff
@@ -1149,6 +1149,24 @@ locally :: MonadReader s m => ASetter s s a b -> (a -> b) -> m r -> m r
 locally l f = Reader.local (over l f)
 {-# INLINE locally #-}
 
+-- | This is a generalization of 'locally' that allows one to make indexed
+-- 'local' changes to a 'Reader' environment associated with the target of a
+-- 'Setter', 'Lens', or 'Traversal'.
+--
+-- @
+-- 'locally' l f ≡ 'ilocally' l f . const
+-- 'ilocally' l f ≡ 'locally' l f . 'Indexed'
+-- @
+--
+-- @
+-- ilocally :: MonadReader s m => 'IndexedLens' s s a b      -> (a -> b) -> m r -> m r
+-- ilocally :: MonadReader s m => 'IndexedTraversal' s s a b -> (a -> b) -> m r -> m r
+-- ilocally :: MonadReader s m => 'IndexedSetter' s s a b    -> (a -> b) -> m r -> m r
+-- @
+ilocally :: MonadReader s m => IndexedSetter i s s a b -> (i -> a -> b) -> m r -> m r
```

Comment:
It should take `AnIndexedSetter`, see https://hackage.haskell.org/package/lens-4.17/docs/Control-Lens-Combinators.html#v:iover

Reply:
Thanks - late night over sight 😄